examples/get_token.rs

```diff
@@ -0,0 +1,37 @@
+use std::env;
+use tokio_core::reactor::Core;
+
+use librespot::core::authentication::Credentials;
+use librespot::core::config::SessionConfig;
+use librespot::core::keymaster;
+use librespot::core::session::Session;
+
+const SCOPES: &str =
+    "streaming,user-read-playback-state,user-modify-playback-state,user-read-currently-playing";
+
+fn main() {
+    let mut core = Core::new().unwrap();
+    let handle = core.handle();
+
+    let session_config = SessionConfig::default();
+
+    let args: Vec<_> = env::args().collect();
+    if args.len() != 4 {
+        println!("Usage: {} USERNAME PASSWORD CLIENT_ID", args[0]);
+    }
+    let username = args[1].to_owned();
+    let password = args[2].to_owned();
+    let client_id = &args[3];
+
+    println!("Connecting..");
+    let credentials = Credentials::with_password(username, password);
+    let session = core
+        .run(Session::connect(session_config, credentials, None, handle))
+        .unwrap();
+
+    println!(
+        "Token: {:#?}",
```

Comment:
Wouldn't that print something like this with a duplicate `Token` prefix?

```
Token: Token {
    access_token: "BQCfSS6x...",
```

Could this easily be changed to print raw, parsable JSON instead? As the keys aren't quoted, parsers might fail.

Reply:
This was is just meant to be an example on how to use the API, JSON is doable, but then I would rather call the mercury (hm) endpoint directly that spits out JSON.
Or you could always manually build up the JSON string with `format!`. 


This would be the "correct" way to do it
```diff
diff --git a/Cargo.lock b/Cargo.lock
index 533b47d..ec25bbd 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -951,6 +951,7 @@ dependencies = [
  "protobuf 2.14.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "rand 0.7.3 (registry+https://github.com/rust-lang/crates.io-index)",
  "rpassword 3.0.2 (registry+https://github.com/rust-lang/crates.io-index)",
+ "serde_json 1.0.56 (registry+https://github.com/rust-lang/crates.io-index)",
  "sha-1 0.8.2 (registry+https://github.com/rust-lang/crates.io-index)",
  "tokio-core 0.1.17 (registry+https://github.com/rust-lang/crates.io-index)",
  "tokio-io 0.1.13 (registry+https://github.com/rust-lang/crates.io-index)",
diff --git a/Cargo.toml b/Cargo.toml
index e2dfda2..4fa1ec9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,6 +58,9 @@ url = "1.7"
 sha-1 = "0.8"
 hex = "0.3"

+[dev-dependencies]
+serde_json = "1"
+
 [features]
 alsa-backend = ["librespot-playback/alsa-backend"]
 portaudio-backend = ["librespot-playback/portaudio-backend"]
diff --git a/core/src/keymaster.rs b/core/src/keymaster.rs
index f2d7b77..4ac55b9 100644
--- a/core/src/keymaster.rs
+++ b/core/src/keymaster.rs
@@ -4,7 +4,7 @@ use serde_json;
 use crate::mercury::MercuryError;
 use crate::session::Session;

-#[derive(Deserialize, Debug, Clone)]
+#[derive(Deserialize, Debug, Clone, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Token {
     pub access_token: String,
diff --git a/examples/get_token.rs b/examples/get_token.rs
index d722e99..0a9761b 100644
--- a/examples/get_token.rs
+++ b/examples/get_token.rs
@@ -5,6 +5,7 @@ use librespot::core::authentication::Credentials;
 use librespot::core::config::SessionConfig;
 use librespot::core::keymaster;
 use librespot::core::session::Session;
+use serde_json;

 const SCOPES: &str =
     "streaming,user-read-playback-state,user-modify-playback-state,user-read-currently-playing";
@@ -28,10 +29,8 @@ fn main() {
     let session = core
         .run(Session::connect(session_config, credentials, None, handle))
         .unwrap();
-
-    println!(
-        "Token: {:#?}",
-        core.run(keymaster::get_token(&session, &client_id, SCOPES))
-            .unwrap()
-    );
+    let token = core
+        .run(keymaster::get_token(&session, &client_id, SCOPES))
+        .unwrap();
+    println!("{:#?}", serde_json::to_string(&token).unwrap());
 }
```

Reply:
Ok, understood, thanks. But my comment about the double "Token" prefix still is true, isn't it?

Reply:
Not sure I follow, the original example prints the `Token` struct using the Debug trait. This isn't necessary valid JSON..

Are you suggesting to remove the `Token:` prefix in the string printed to stdout to make it easier to parse?

Reply:
I believe what I'm seeing is that if you dump the `Token` struct, that word `Token` would already be part of the dump. No need to print it in the `println` statement (but then I don't have the tools here to double check).

Reply:
Ah, I was just following the `variable: {value}` nomenclature. Especially given this is an example, it should make things explicit?